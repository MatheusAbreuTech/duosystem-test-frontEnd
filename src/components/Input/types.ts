import { InputHTMLAttributes } from 'react'

export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  name: string
  dataTestid: string
  type?: string
  id?: string
}
