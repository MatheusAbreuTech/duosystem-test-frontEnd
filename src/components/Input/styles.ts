import theme from '@/styles/theme'
import styled from 'styled-components'

export const Input = styled.input`
  width: 100%;
  height: 42px;
  padding: 0 1rem;
  background-color: #fff;
  outline: none;
  border: 1px solid ${theme.colors.black};
  font-size: 1.6rem;
  border-radius: 5px;
`
