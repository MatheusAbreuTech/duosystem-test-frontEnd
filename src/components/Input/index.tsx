import * as S from './styles'
import { useFormContext } from 'react-hook-form'
import { InputProps } from './types'

const Input = ({ name, type, id, dataTestid }: InputProps) => {
  const { register } = useFormContext()

  return (
    <S.Input
      type={type && type}
      id={id && id}
      data-testid={dataTestid && dataTestid}
      {...register(name)}
    />
  )
}

export default Input
