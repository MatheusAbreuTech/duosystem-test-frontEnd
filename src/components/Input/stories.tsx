import { Meta, StoryFn } from '@storybook/react'
import Input from '.'
import { FormProvider, useForm } from 'react-hook-form'
import { SchemaType } from '../AddTask/types'
import { zodResolver } from '@hookform/resolvers/zod'
import { schema } from '../AddTask/schema'

export default {
  title: 'Input',
  component: Input,
  argTypes: {
    onChange: { action: 'onChange' }
  }
} as Meta

const Template: StoryFn = () => {
  const methods = useForm<SchemaType>({
    resolver: zodResolver(schema)
  })

  return (
    <FormProvider {...methods}>
      <Input name="taskDescription" />
    </FormProvider>
  )
}

export const Default = Template.bind({})
