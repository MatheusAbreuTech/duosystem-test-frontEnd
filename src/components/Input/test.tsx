import { fireEvent, render } from '@testing-library/react'

import Input from '.'
import { FormProvider, useForm } from 'react-hook-form'
import { zodResolver } from '@hookform/resolvers/zod'
import { z } from 'zod'

const schema = z.object({
  test: z.string()
})
type ValidationSchemaType = z.infer<typeof schema>

describe('<Input />', () => {
  test('should renders with default values', () => {
    const TestComponent = () => {
      const methods = useForm<ValidationSchemaType>({
        resolver: zodResolver(schema),
        defaultValues: {
          test: 'teste'
        }
      })

      return (
        <FormProvider {...methods}>
          <Input name="test" />
        </FormProvider>
      )
    }

    const { container } = render(<TestComponent />)

    const inputElement = container.querySelector('#test') as HTMLInputElement

    expect(inputElement).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  test('should change value when type', () => {
    const TestComponent = () => {
      const methods = useForm<ValidationSchemaType>({
        resolver: zodResolver(schema)
      })

      return (
        <FormProvider {...methods}>
          <Input name="test" />
        </FormProvider>
      )
    }

    const { container } = render(<TestComponent />)

    const inputElement = container.querySelector('#test') as HTMLInputElement

    fireEvent.change(inputElement, { target: { value: 'Novo Valor' } })

    const inputValue = inputElement.value

    expect(inputValue).toBe('Novo Valor')
  })
})
