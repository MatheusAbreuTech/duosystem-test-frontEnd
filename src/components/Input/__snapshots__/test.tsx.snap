// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should renders with default values 1`] = `
.c0 {
  width: 100%;
  height: 42px;
  padding: 0 1rem;
  background-color: #fff;
  outline: none;
  border: none;
  font-size: 1.6rem;
  border-radius: 5px;
}

<input
  class="c0"
  id="test"
  name="test"
/>
`;
